dask/dataframe/core.py

```diff
@@ -2959,11 +2957,18 @@ def quantile(self, q=0.5, axis=0, numeric_only=no_default, method="default"):
                 **numeric_kwargs,
                 meta=(q, "f8"),
                 parent_meta=self._meta,
+                _dask_method_name="quantile",
             )
         else:
             _raise_if_object_series(self, "quantile")
-            num = self._get_numeric_data()
-            quantiles = tuple(quantile(self[c], q, method) for c in num.columns)
+            num = (
+                self
+                if PANDAS_GT_200
+                and not numeric_kwargs.get("numeric_only", False)
```

Comment:
This is an odd-looking condition, it's hard to read:

```python
and not numeric_kwargs.get("numeric_only", False)
or not numeric_kwargs.get("numeric_only", True)
```

Are you looking for `numeric_only in (False, no_default)`?

Reply:
No not that simple unfortunately. We have to adjust for the changed default in pandas. 

I rewrote the condition, let me know whether this is better

Reply:
How about this? Just a little bit simpler.

```
num = (
    self.get_numeric_data()
    if numeric_only is True
    or (not PANDAS_GT_200 and numeric_only is no_default)
    else self
)
```

Reply:
changed, thx